modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -150,7 +151,7 @@ public class ReplicaManager extends AbstractEventProducer<LocalReplicaEvent, Loc
     private static final PlacementDriverMessagesFactory PLACEMENT_DRIVER_MESSAGES_FACTORY = new PlacementDriverMessagesFactory();
 
     /** Busy lock to stop synchronously. */
-    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+    private final IgniteStripedReadWriteLock busyLock = new IgniteStripedReadWriteLock();
```

Comment:
Same as above, the contract is messy. ReplicaManager is not an utility class, thus it's unlikely that stopAsync will be called in the same thread that will also acquire readLocks. However, from the busyLock's general perspective, it's still a violation.

Reply:
https://issues.apache.org/jira/browse/IGNITE-23716